### PR TITLE
KOGITO-3372 - Validate explainability requests and add tests

### PR DESCRIPTION
**Pull request by r00ta**
https://issues.redhat.com/browse/KOGITO-3372

Some fields of the explainability rest requests are mandatory, this PR aims to generate a bad request in such cases instead of making the application to crash with status 500.

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: explainability/explainability-service-rest/src/main/java/org/kie/kogito/explainability/rest/ExplainabilityApiV1.java [hunk 1]
```diff
@@ -53,6 +55,10 @@ public ExplainabilityApiV1(
         this.predictionProviderFactory = predictionProviderFactory;
     }
 
+    private boolean isNullOrEmpty(String s) {
+        return s == null || s == "";
+    }
```
Comment (danielezonca) [inline explainability/explainability-service-rest/src/main/java/org/kie/kogito/explainability/rest/ExplainabilityApiV1.java]: What about use [StringUtils](http://commons.apache.org/proper/commons-lang/javadocs/api-3.1/org/apache/commons/lang3/StringUtils.html)? If you don't want to add the dependency please at least move the method to a local `StringUtils` class

File: explainability/explainability-service-rest/src/main/java/org/kie/kogito/explainability/rest/ExplainabilityApiV1.java [hunk 2]
```diff
@@ -64,12 +70,41 @@ public ExplainabilityApiV1(
...
+        if (modelIdentifierDto == null || isNullOrEmpty(modelIdentifierDto.getResourceType()) || isNullOrEmpty(modelIdentifierDto.getResourceId())) {
+            return new RequestValidationResult(false, "The model identifier information is required in the request.");
+        }
+
+        if (isNullOrEmpty(requestDto.getServiceUrl())) {
+            return new RequestValidationResult(false, "The service url information of the application that evaluated the decision is not provided in the request.");
+        }
+
+        return new RequestValidationResult(true, null);
+    }
```
Comment (danielezonca) [inline explainability/explainability-service-rest/src/main/java/org/kie/kogito/explainability/rest/ExplainabilityApiV1.java]: What about move it to a util class?

**Comment by Sgitario** [comment]
@r00ta why you didn't use the validation extension: https://quarkus.io/guides/validation ? I think we can achieve the same with much lesser code :/

**Comment by r00ta** [comment]
Hi @Sgitario , thanks for the tip! Let me know wdyt now

File: explainability/explainability-service-rest/src/main/java/org/kie/kogito/explainability/rest/ExplainabilityApiV1.java [hunk 3]
```diff
@@ -64,6 +72,16 @@ public ExplainabilityApiV1(
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.APPLICATION_JSON)
     public Uni<Response> explain(ExplainabilityRequestDto requestDto) {
+        Set<ConstraintViolation<ExplainabilityRequestDto>> violations = validator.validate(requestDto);
```
Comment (Sgitario) [inline explainability/explainability-service-rest/src/main/java/org/kie/kogito/explainability/rest/ExplainabilityApiV1.java]: As far I know, the annotation @Valid works fine with reactive endpoints, so if we annotate the requestDto field with @Valid, it should raise 400 Invalid Request directly.

**Review by Sgitario — COMMENTED** [review]
It looks much better now. I've just asked why you used the validator service instead of annotating the field in the method with "@Valid".

File: explainability/explainability-service-rest/src/main/java/org/kie/kogito/explainability/rest/ExplainabilityApiV1.java (hunk 3 above)
Comment (r00ta) [reply explainability/explainability-service-rest/src/main/java/org/kie/kogito/explainability/rest/ExplainabilityApiV1.java]: Thanks @Sgitario , this module was new for me, good to know :) lmkwyt

File: explainability/explainability-service-rest/src/main/java/org/kie/kogito/explainability/rest/ExplainabilityApiV1.java [hunk 4]
```diff
@@ -44,13 +46,16 @@
 
     protected ExplanationService explanationService;
     protected PredictionProviderFactory predictionProviderFactory;
+    protected Validator validator;
```
Comment (Sgitario) [inline explainability/explainability-service-rest/src/main/java/org/kie/kogito/explainability/rest/ExplainabilityApiV1.java]: I think this service is a leftover

**Review by Sgitario — APPROVED** [review]
lgtm now. Just a leftover. Thanks!

**Review by danielezonca — APPROVED** [review]
LGTM 👍

**Pull request merged**